lib/fluent/plugin/filter_process_snmptrap.rb

```diff
@@ -38,31 +39,45 @@ def start
     end
 
     def filter(tag, time, record)
+      @time = time
+      @tag = tag
       message = record.to_s
       message = message.delete('\\"')
       snmp_msg = message.gsub(/(?:(SNMPv2-(\w+)(::)(\w+)((\.)(\d+)){1,13}(=>))|(host=>))/, "")
       record["machineId"] = ""
-      record["rmc_host"] = ""
+      record["rmcHostIP"] = ""
       record["event"] = ""
       record["status"] = ""
-      record["sensor"] = ""
+      record["device"] = ""
       record["severity"] = ""
       record["sensorValue"] = ""
       record["error"] = ""
       record["message"] = ""
       record["timestamp"] = ""
 
       determineMachineId(record)
-      getrmchost(record)
+      getrmcHost(record)
       processEvent(record)
       determineSensorValue(record)
       determineStatus(record)
-      record["message"] = snmp_msg
+      makeMessage(record)
       record["timestamp"] = time
       record.delete_if { |key, value| key.to_s.match(/(?:SNMPv2-(\w+)(::)(\w+)((\.)(\d+)){1,13}|(host))/)}
       return record
     end
 
+    def makeMessage(record)
+      message = 
+      {
+        :timestamp => @time,
```

Comment:
why the message has the timestamp again? isn't it already on `record["timestamp"]` ?

Reply:
yes, removed